### ISSUE TITLE
System.out cannot be resolved to a type in JSPs in Java 17

### ISSUE THREAD

**Issue by gjwatts** [OpenLiberty/open-liberty#18043]
When trying to serve up a Java Server Pages (JSPs) in Java 17 (currently using build 33) that uses `System.out`, for example:

```
<html>
<title>updateable.jsp</title>
<body>
	<%
		String greeting = request.getParameter("greeting");
		String userName = "Joe";
		System.out.println("executing updateable.jsp - printing " + greeting + " " + userName);
	%>
	<div><%=greeting%> <%=userName%></div>
</body>
</html>
```

the following error is seen:
```
...
System.out cannot be resolved to a type<BR>
at com.ibm.ws.jsp.webcontainerext.AbstractJSPExtensionServletWrapper.translateJsp&#40;AbstractJSPExtensionServletWrapper.java:603&#41;<BR>
at com.ibm.ws.jsp.webcontainerext.AbstractJSPExtensionServletWrapper._checkForTranslation&#40;AbstractJSPExtensionServletWrapper.java:463&#41;<BR>
at com.ibm.ws.jsp.webcontainerext.AbstractJSPExtensionServletWrapper.checkForTranslation&#40;AbstractJSPExtensionServletWrapper.java:253&#41;<BR>
at com.ibm.ws.jsp.webcontainerext.AbstractJSPExtensionProcessor.findWrapper&#40;AbstractJSPExtensionProcessor.java:512&#41;<BR>
at com.ibm.ws.jsp.webcontainerext.AbstractJSPExtensionProcessor.getServletWrapper&#40;AbstractJSPExtensionProcessor.java:355&#41;<BR> 
at com.ibm.ws.webcontainer.webapp.WebApp.handleRequest&#40;WebApp.java:5019&#41;<BR> 
at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost$2.handleRequest&#40;DynamicVirtualHost.java:314&#41;<BR> 
at com.ibm.ws.webcontainer.WebContainer.handleRequest&#40;WebContainer.java:1007&#41;<BR> 
at com.ibm.ws.webcontainer.osgi.DynamicVirtualHost$2.run&#40;DynamicVirtualHost.java:279&#41;<BR> 
at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink$TaskWrapper.run&#40;HttpDispatcherLink.java:1159&#41;<BR> 
at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.wrapHandlerAndExecute&#40;HttpDispatcherLink.java:428&#41;<BR> 
at com.ibm.ws.http.dispatcher.internal.channel.HttpDispatcherLink.ready&#40;HttpDispatcherLink.java:387&#41;<BR> 
at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleDiscrimination&#40;HttpInboundLink.java:566&#41;<BR> 
at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.handleNewRequest&#40;HttpInboundLink.java:500&#41;<BR> 
at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.processRequest&#40;HttpInboundLink.java:360&#41;<BR> 
at com.ibm.ws.http.channel.internal.inbound.HttpInboundLink.ready&#40;HttpInboundLink.java:327&#41;<BR> 
at com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.sendToDiscriminators&#40;NewConnectionInitialReadCallback.java:167&#41;<BR> 
at com.ibm.ws.tcpchannel.internal.NewConnectionInitialReadCallback.complete&#40;NewConnectionInitialReadCallback.java:75&#41;<BR> 
at com.ibm.ws.tcpchannel.internal.WorkQueueManager.requestComplete&#40;WorkQueueManager.java:504&#41;<BR> 
at com.ibm.ws.tcpchannel.internal.WorkQueueManager.attemptIO&#40;WorkQueueManager.java:574&#41;<BR> 
at com.ibm.ws.tcpchannel.internal.WorkQueueManager.workerRun&#40;WorkQueueManager.java:958&#41;<BR> 
at com.ibm.ws.tcpchannel.internal.WorkQueueManager$Worker.run&#40;WorkQueueManager.java:1047&#41;<BR> 
at com.ibm.ws.threading.internal.ExecutorServiceImpl$RunnableWrapper.run&#40;ExecutorServiceImpl.java:238&#41;<BR> 
at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker&#40;ThreadPoolExecutor.java:1136&#41;<BR> 
at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run&#40;ThreadPoolExecutor.java:635&#41;<BR> 
at java.base/java.lang.Thread.run&#40;Thread.java:833&#41;
...
```

And causes almost 30 of our FATs to fail.  This error is not seen in Java 16 or earlier.

Looking over the Java 17 [features](https://openjdk.java.net/projects/jdk/17/) and [release notes](https://jdk.java.net/17/release-notes), nothing explicit can be found about changes to `System.out` that would cause this behavior.

**Comment by epj** [comment]
Adding <%@page import="java.io.PrintStream"%> to the JSP avoids the error, but we need to determine why this became necessary.  Something in Java 17, or how the classes are accessed by JSP, changed.

**Comment by gjwatts** [comment]
I have reported this as a potential bug to Oracle.

**Comment by pnicolucci** [comment]
@gjwatts  do you have a link to the bug?

**Comment by gjwatts** [comment]
@pnicolucci I am not sure if you are referring to our bug or the bug I reported to Oracle

Ours (sorry for the IBM only link) -> [here](https://wasrtc.hursley.ibm.com:9443/jazz/web/projects/WS-CD#action=com.ibm.team.build.viewResult&id=_o7MGoPVbEeuj_9vf1o2UsA&tab=com.ibm.team.build.web.ui.internal.editors.result.test.TestProvider) for an example (look at the first failing test for com.ibm.ws.classloading_fat)

Oracle -> [here](https://bugs.java.com/bugdatabase/view_bug.do?bug_id=JDK-8272079)

I tested a similar setup in Apache Tomcat 9.0.50 and it worked fine in the same version of Java 17 for me.  Currently looks like this is our issue.

**Comment by volosied** [comment]
I agree -- I also just tested on Tomcat (slightly older 9.0.29) and the JSP worked successfully. This issue appears to be a bug somewhere in our implementation.  Evan and I will investigate further.

**Comment by gjwatts** [comment]
@volosied @epj Thank you

**Comment by brideck** [comment]
This problem fails over 3000 tests in our Java 17 runs of the Jakarta EE TCK, as JSPs are one of the primary test vehicles.

**Comment by epj** [comment]
At the moment the working theory is that the JDT needs to be updated.  Liberty is currently using a version from 2016.

**Comment by gjwatts** [comment]
@epj thanks for that update.  Any idea how much work is involved and how long it will take to update the JDT?

**Comment by volosied** [comment]
Merging #18189  automatically closed this issue.  However, it should be fixed with the 3.22. PR. 

Feel free to re-opoen if needed.
